### PR TITLE
add max size feature to grover vocabulary

### PR DESCRIPTION
**Pull request by gusty1g**
# Pull Request Template

## Description

Adds a new `max_size` parameter to grover vocabulary which can be used for setting the size of vocabulary.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/feat/vocabulary_builders/grover_vocab.py [hunk 1]
```diff
@@ -25,6 +25,10 @@ class GroverAtomVocabularyBuilder(VocabularyBuilder):
     in the dataset and makes a vocabulary to index mapping by sorting the vocabulary
     by frequency and then alphabetically.
```
Comment (rbharath) [inline deepchem/feat/vocabulary_builders/grover_vocab.py:26]: Can you add a comment explaining what happens when max_size is set in more detail? It looks like the algorithm stops counting new vocab elements once the limit is hit which is reasonable behavior but would be good to document here

**Review by rbharath — COMMENTED** [review]
One minor request for more comments but otherwise good to go

File: deepchem/feat/vocabulary_builders/grover_vocab.py (hunk 1 above)
Comment (gusty1g) [reply deepchem/feat/vocabulary_builders/grover_vocab.py:26]: added documentation

**Review by rbharath — APPROVED** [review]
LGTM, feel free to merge in once CI is clear. There look to be a few CI failures though, but I'm not sure if any of them are related to this PR

**Comment by gusty1g** [comment]
merged as part of #3272

**Pull request closed without merging**